{"uid":"56dfac04315dcb4f","name":"Login with Valid Credentials","fullName":"Flask.tests_Selenium.test_userflow_ui_selenium#test_login","historyId":"fb1335df6add77112c9c2bb0d237512b","time":{"start":1760730792547,"stop":1760730792692,"duration":145},"status":"broken","statusMessage":"selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=141.0.7390.107)\nStacktrace:\n#0 0x558ed9d0167a <unknown>\n#1 0x558ed9780566 <unknown>\n#2 0x558ed9777470 <unknown>\n#3 0x558ed9768926 <unknown>\n#4 0x558ed976a72b <unknown>\n#5 0x558ed9768eb1 <unknown>\n#6 0x558ed9768650 <unknown>\n#7 0x558ed9768302 <unknown>\n#8 0x558ed9766142 <unknown>\n#9 0x558ed976692a <unknown>\n#10 0x558ed9783a99 <unknown>\n#11 0x558ed981eee5 <unknown>\n#12 0x558ed97f8632 <unknown>\n#13 0x558ed981e328 <unknown>\n#14 0x558ed97f8403 <unknown>\n#15 0x558ed97c4b02 <unknown>\n#16 0x558ed97c57c1 <unknown>\n#17 0x558ed9cc5448 <unknown>\n#18 0x558ed9cc92af <unknown>\n#19 0x558ed9cac8d9 <unknown>\n#20 0x558ed9cc9e55 <unknown>\n#21 0x558ed9c9213f <unknown>\n#22 0x558ed9cee4b8 <unknown>\n#23 0x558ed9cee693 <unknown>\n#24 0x558ed9d00613 <unknown>\n#25 0x7f720929caa4 <unknown>\n#26 0x7f7209329c6c <unknown>","statusTrace":"driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"24d5ab244d93bfd6d35c01f857c11acc\")>\n\n    @allure.severity(Severity.CRITICAL)\n    @allure.title(\"Login with Valid Credentials\")\n    def test_login(driver):\n        with allure.step(\"Open Login Page\"):\n>           driver.get(f\"{BASE_URL}/login\")\n\nFlask/tests_Selenium/test_userflow_ui_selenium.py:60: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:479: in get\n    self.execute(Command.GET, {\"url\": url})\n/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:454: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f0fcdc1b100>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"unknown error\",\"message\":\"unknown error: net::ERR_CONNECTION_REFUSED\\\\n  ...\\\\n#24 0x558ed9d00613 \\\\u003Cunknown>\\\\n#25 0x7f720929caa4 \\\\u003Cunknown>\\\\n#26 0x7f7209329c6c \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                try:\n                    value = json.loads(value_json)\n                    if isinstance(value, dict):\n                        if len(value) == 1:\n                            value = value[\"value\"]\n                        status = value.get(\"error\", None)\n                        if not status:\n                            status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                            message = value.get(\"value\") or value.get(\"message\")\n                            if not isinstance(message, str):\n                                value = message\n                                message = message.get(\"message\")\n                        else:\n                            message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE         (Session info: chrome=141.0.7390.107)\nE       Stacktrace:\nE       #0 0x558ed9d0167a <unknown>\nE       #1 0x558ed9780566 <unknown>\nE       #2 0x558ed9777470 <unknown>\nE       #3 0x558ed9768926 <unknown>\nE       #4 0x558ed976a72b <unknown>\nE       #5 0x558ed9768eb1 <unknown>\nE       #6 0x558ed9768650 <unknown>\nE       #7 0x558ed9768302 <unknown>\nE       #8 0x558ed9766142 <unknown>\nE       #9 0x558ed976692a <unknown>\nE       #10 0x558ed9783a99 <unknown>\nE       #11 0x558ed981eee5 <unknown>\nE       #12 0x558ed97f8632 <unknown>\nE       #13 0x558ed981e328 <unknown>\nE       #14 0x558ed97f8403 <unknown>\nE       #15 0x558ed97c4b02 <unknown>\nE       #16 0x558ed97c57c1 <unknown>\nE       #17 0x558ed9cc5448 <unknown>\nE       #18 0x558ed9cc92af <unknown>\nE       #19 0x558ed9cac8d9 <unknown>\nE       #20 0x558ed9cc9e55 <unknown>\nE       #21 0x558ed9c9213f <unknown>\nE       #22 0x558ed9cee4b8 <unknown>\nE       #23 0x558ed9cee693 <unknown>\nE       #24 0x558ed9d00613 <unknown>\nE       #25 0x7f720929caa4 <unknown>\nE       #26 0x7f7209329c6c <unknown>\n\n/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:232: WebDriverException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"driver","time":{"start":1760730792294,"stop":1760730792546,"duration":252},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"status":"broken","statusMessage":"selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=141.0.7390.107)\nStacktrace:\n#0 0x558ed9d0167a <unknown>\n#1 0x558ed9780566 <unknown>\n#2 0x558ed9777470 <unknown>\n#3 0x558ed9768926 <unknown>\n#4 0x558ed976a72b <unknown>\n#5 0x558ed9768eb1 <unknown>\n#6 0x558ed9768650 <unknown>\n#7 0x558ed9768302 <unknown>\n#8 0x558ed9766142 <unknown>\n#9 0x558ed976692a <unknown>\n#10 0x558ed9783a99 <unknown>\n#11 0x558ed981eee5 <unknown>\n#12 0x558ed97f8632 <unknown>\n#13 0x558ed981e328 <unknown>\n#14 0x558ed97f8403 <unknown>\n#15 0x558ed97c4b02 <unknown>\n#16 0x558ed97c57c1 <unknown>\n#17 0x558ed9cc5448 <unknown>\n#18 0x558ed9cc92af <unknown>\n#19 0x558ed9cac8d9 <unknown>\n#20 0x558ed9cc9e55 <unknown>\n#21 0x558ed9c9213f <unknown>\n#22 0x558ed9cee4b8 <unknown>\n#23 0x558ed9cee693 <unknown>\n#24 0x558ed9d00613 <unknown>\n#25 0x7f720929caa4 <unknown>\n#26 0x7f7209329c6c <unknown>","statusTrace":"driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"24d5ab244d93bfd6d35c01f857c11acc\")>\n\n    @allure.severity(Severity.CRITICAL)\n    @allure.title(\"Login with Valid Credentials\")\n    def test_login(driver):\n        with allure.step(\"Open Login Page\"):\n>           driver.get(f\"{BASE_URL}/login\")\n\nFlask/tests_Selenium/test_userflow_ui_selenium.py:60: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:479: in get\n    self.execute(Command.GET, {\"url\": url})\n/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:454: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f0fcdc1b100>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"unknown error\",\"message\":\"unknown error: net::ERR_CONNECTION_REFUSED\\\\n  ...\\\\n#24 0x558ed9d00613 \\\\u003Cunknown>\\\\n#25 0x7f720929caa4 \\\\u003Cunknown>\\\\n#26 0x7f7209329c6c \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                try:\n                    value = json.loads(value_json)\n                    if isinstance(value, dict):\n                        if len(value) == 1:\n                            value = value[\"value\"]\n                        status = value.get(\"error\", None)\n                        if not status:\n                            status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                            message = value.get(\"value\") or value.get(\"message\")\n                            if not isinstance(message, str):\n                                value = message\n                                message = message.get(\"message\")\n                        else:\n                            message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE         (Session info: chrome=141.0.7390.107)\nE       Stacktrace:\nE       #0 0x558ed9d0167a <unknown>\nE       #1 0x558ed9780566 <unknown>\nE       #2 0x558ed9777470 <unknown>\nE       #3 0x558ed9768926 <unknown>\nE       #4 0x558ed976a72b <unknown>\nE       #5 0x558ed9768eb1 <unknown>\nE       #6 0x558ed9768650 <unknown>\nE       #7 0x558ed9768302 <unknown>\nE       #8 0x558ed9766142 <unknown>\nE       #9 0x558ed976692a <unknown>\nE       #10 0x558ed9783a99 <unknown>\nE       #11 0x558ed981eee5 <unknown>\nE       #12 0x558ed97f8632 <unknown>\nE       #13 0x558ed981e328 <unknown>\nE       #14 0x558ed97f8403 <unknown>\nE       #15 0x558ed97c4b02 <unknown>\nE       #16 0x558ed97c57c1 <unknown>\nE       #17 0x558ed9cc5448 <unknown>\nE       #18 0x558ed9cc92af <unknown>\nE       #19 0x558ed9cac8d9 <unknown>\nE       #20 0x558ed9cc9e55 <unknown>\nE       #21 0x558ed9c9213f <unknown>\nE       #22 0x558ed9cee4b8 <unknown>\nE       #23 0x558ed9cee693 <unknown>\nE       #24 0x558ed9d00613 <unknown>\nE       #25 0x7f720929caa4 <unknown>\nE       #26 0x7f7209329c6c <unknown>\n\n/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:232: WebDriverException","steps":[{"name":"Open Login Page","time":{"start":1760730792547,"stop":1760730792691,"duration":144},"status":"broken","statusMessage":"selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=141.0.7390.107)\nStacktrace:\n#0 0x558ed9d0167a <unknown>\n#1 0x558ed9780566 <unknown>\n#2 0x558ed9777470 <unknown>\n#3 0x558ed9768926 <unknown>\n#4 0x558ed976a72b <unknown>\n#5 0x558ed9768eb1 <unknown>\n#6 0x558ed9768650 <unknown>\n#7 0x558ed9768302 <unknown>\n#8 0x558ed9766142 <unknown>\n#9 0x558ed976692a <unknown>\n#10 0x558ed9783a99 <unknown>\n#11 0x558ed981eee5 <unknown>\n#12 0x558ed97f8632 <unknown>\n#13 0x558ed981e328 <unknown>\n#14 0x558ed97f8403 <unknown>\n#15 0x558ed97c4b02 <unknown>\n#16 0x558ed97c57c1 <unknown>\n#17 0x558ed9cc5448 <unknown>\n#18 0x558ed9cc92af <unknown>\n#19 0x558ed9cac8d9 <unknown>\n#20 0x558ed9cc9e55 <unknown>\n#21 0x558ed9c9213f <unknown>\n#22 0x558ed9cee4b8 <unknown>\n#23 0x558ed9cee693 <unknown>\n#24 0x558ed9d00613 <unknown>\n#25 0x7f720929caa4 <unknown>\n#26 0x7f7209329c6c <unknown>\n\n","statusTrace":"  File \"/home/runner/work/GithubPortfolio/GithubPortfolio/Flask/tests_Selenium/test_userflow_ui_selenium.py\", line 60, in test_login\n    driver.get(f\"{BASE_URL}/login\")\n  File \"/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 479, in get\n    self.execute(Command.GET, {\"url\": url})\n  File \"/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 454, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.10.19/x64/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py\", line 232, in check_response\n    raise exception_class(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"86f8a7f69bcd2c3e","name":"Screenshot","source":"86f8a7f69bcd2c3e.png","type":"image/png","size":30008}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":1,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"driver::0","time":{"start":1760730792806,"stop":1760730792865,"duration":59},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"severity","value":"critical"},{"name":"parentSuite","value":"Flask.tests_Selenium"},{"name":"suite","value":"test_userflow_ui_selenium"},{"name":"host","value":"runnervmzdgdc"},{"name":"thread","value":"3007-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"Flask.tests_Selenium.test_userflow_ui_selenium"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"critical","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"tags":[]},"source":"56dfac04315dcb4f.json","parameterValues":[]}